File: skellig-feature/src/main/java/org/skellig/feature/ParametersUtils.java
package org.skellig.feature;

import java.util.Map;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

class ParametersUtils {

    private static final Pattern PARAM_REGEX = Pattern.compile("<([\\w_-]+)>");

    static String replaceParametersIfFound(String value, Map<String, String> data) {
        Matcher matcher = PARAM_REGEX.matcher(value);
        while (matcher.find()) {
            String paramNameWithBrackets = matcher.group(0);
            String paramName = matcher.group(1);
            String paramValue = data.getOrDefault(paramName, "");
            value = value.replace(paramNameWithBrackets, paramValue);
        }
        return value;
    }
}

File: skellig-feature/src/main/kotlin/org/skellig/feature/ParametersUtils.kt
package org.skellig.feature

import java.util.regex.Pattern

internal class ParametersUtils {
    companion object {
        private val PARAM_REGEX = Pattern.compile("<([\\w_-]+)>")

        /**
         * Replaces parameter placeholders in the given value with corresponding values from the data map.
         *
         * @param value The string value with parameter placeholders to be replaced.
         * @param data The map containing parameter values.
         * @return The updated string with replaced parameter values.
         */
        fun replaceParametersIfFound(value: String, data: Map<String, String?>): String {
            var newValue = value
            val matcher = PARAM_REGEX.matcher(newValue)
            while (matcher.find()) {
                val paramNameWithBrackets = matcher.group(0)
                val paramName = matcher.group(1)
                val paramValue = data.getOrDefault(paramName, "")
                newValue = newValue.replace(paramNameWithBrackets, paramValue!!)
            }
            return newValue
        }
    }
}